src/main/java/com/netcracker/study/general/GameField.java

```diff
@@ -1,12 +1,99 @@
 package com.netcracker.study.general;
 
+import com.netcracker.study.objects.borders.BorderObject;
+import com.netcracker.study.objects.borders.Wall;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.util.ArrayList;
+import java.util.Scanner;
+
 public class GameField {
     private int width;
     private int height;
+    private BorderObject[] borders;
 
 
     public GameField(int width, int height) {
         this.width = width;
         this.height = height;
+        borders = new BorderObject[totalCountOfBorders(width, height)];
+    }
+
+    public GameField(String mapFilePath) throws FileNotFoundException {
+        try{
+            File mapFile = new File(mapFilePath);
+            FileReader input = new FileReader(mapFile);
+            Scanner scan = new Scanner(input);
+            ArrayList<String> mapLines = new ArrayList<String>();
+            while (scan.hasNextLine()){
+                mapLines.add(scan.nextLine());
+            }
+            this.width = (mapLines.get(0).length()-1)/2;
+            this.height = (mapLines.size()-1)/2;
+
+            for (int i = mapLines.size()-1; i > 0; i = i - 2){
+
+            }
+        } catch (FileNotFoundException e) {
+            throw new FileNotFoundException("Map file not found!");
+        }
+    }
+
+    private int totalCountOfBorders(int width, int height){
+        return ((2*width+1)*height + width);
+    }
+
+    private Integer getBorderBetweenTiles(int x1, int x2, int y1, int y2) throws IllegalArgumentException {
+
+        if (Math.abs(x1 - x2) > 1 || Math.abs(y1 - y2) > 1 || (x1 - x2 != 0 && y1 - y2 != 0)) {
+            throw new IllegalArgumentException("Tiles selected are not adjacent => border cannot be specified!");
+        }
+        if (x1 - x2 == 0 && y1 - y2 == 0) {
```

Comment:
all these if sections should be extracted to private methods. This is very hard to read